Add isModeEnabledFn to zoom options

davidswinegar opened the pull request:
Add an optional function property, isModeEnabledFn to the zoom
options. This allows disabling modes based on other keyboard
properties.
Ex. disabling the 'y' direction when a particular key is pressed

davidswinegar committed:
Add isModeEnabledFn to zoom options

Add an optional function property, isModeEnabledFn to the zoom
options. This allows disabling modes based on other keyboard
properties.
Ex. disabling the 'y' direction when a particular key is pressed

vidaaudrey commented on README.md:
```diff
@@ -68,6 +68,13 @@ plugins: {
+			// Ex. disabling the 'y' direction when a particular key is pressed.
+			// event: nullable DOM event
+			// direction: 'x' or 'y'
+			isModeEnabledFn: (event, direction) => true,
```
I wonder if there is a simpler approach as it's a little complicate to understand the logic, e.g. if `mode` is `xy`, and `isModeEnabledFn` returns false, which mode should `mode` be?

what about changing `mode` itself to be either the mode string or a function that returns the mode string?

davidswinegar replied:
I thought about that but I also wanted to rewrite as little of this library as possible :) The `isModeEnabledFn` overrides the `mode`, but only for disabling a particular mode

davidswinegar replied:
@vidaaudrey I'm going to rename this to `isModeEnabledOverrideFn` for clarity that it's an override

vidaaudrey replied:
We still want to merge this into the official package. Would be good to create a PR against `chartjs/chartjs-plugin-zoom` and get it merged there. The maintainer might have some additional feedbacks.

rdgifford reviewed (approved):
To @vidaaudrey's point, you could do a map of isModeEnabledFns to modes. Then you know that for each mode you have one code path. Honestly though, this seems fine to me. The only modes are x and y, right?

davidswinegar committed:
Add built files to distribute via github

Need to commit these since we are distributing this package via
github, not through NPM.
